odoo/addons/base/models/ir_qweb.py

```diff
@@ -618,9 +619,12 @@ def _get_view_id(self, template):
 
     @QwebTracker.wrap_compile
     def _compile(self, template):
+        ref = None
         if isinstance(template, etree._Element):
             self = self.with_context(is_t_cache_disabled=True)
-            ref = None
+        elif isinstance(template, str) and template.endswith('.xml'):
+            with file_open(template, 'rb') as file:
```

Comment:
```suggestion
            with file_open(template, 'rb', filter_ext=('.xml',)) as file:
```

Extra security precaution, currently it's not useful as there is already the condition above `template.endswith('.xml')`, but it will make things future-proof in case the code moves and/or this condition changes.

Reply:
the test `test_message_error_3` tries to render a template as a string like `"<document></document>"` instead of an etree. So it is usefull in that case as it should not change the current behaviour.

Reply:
dle just ask to add the extra check `filter_ext=('.xml',)` in the file_open call ;)  (while keeping the other extension check at the same time)

Reply:
I am worried about exposing file reading in the `_compile` method and thus `_render`. A few years ago I had to disallow providing a template or url for security reasons. @beledouxdenis  has more experience in security.

For example via this public route, if there is sensitive data in the `security` directory, someone could have access to it.
```py
iap_authorize(env, key, accou..., credit_template=None, ...)
```

Couldn't the idea of ​​putting a `from_file` flag on views be used ? When it is true, the view will look on the system file (from the `arch_fs` data). Only the xml would be used, and that covers your needs without any security risk.

Reply:
> Could the idea of ​​putting a from_file flag on views be used ?

If I am not mistaken, the path of the file is stored in database, so you could change that path to an arbitrary one and it would result in the same vulnerability.

As Xavier suggested if we want to make it super secure we can take a policy that the file as to be an XML file inside a specific sub-folder of the module, such as `templates` or something.
That way you cannot read arbitrary file, only `/*/templates/*.xml`, amd we guarantee there are no secrets in there.

Reply:
I initially intended to do use a subfolder 'documents' but it was removed. I don't remember the reason.